tensorflow_model_optimization/python/core/clustering/keras/cluster.py

```diff
@@ -197,6 +197,7 @@ def _strip_clustering_wrapper(layer):
       for i in range(len(layer.restore)):
         # This is why we used integers as keys
         name, weight = layer.restore[i]
```

Comment:
Please, consider storing the correct name in "restore" in the first place. It only purpose is to reconstruct the original names here so that we may as well save the correct values when filling it in cluster_wrapper.py.